grafana/resource_machine_learning_holiday.go

```diff
@@ -0,0 +1,202 @@
+			]
+		}
+`,
+
+		CreateContext: resourceMachineLearningHolidayCreate,
+		ReadContext:   resourceMachineLearningHolidayRead,
+		UpdateContext: resourceMachineLearningHolidayUpdate,
+		DeleteContext: resourceMachineLearningHolidayDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Description: "The ID of the holiday.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"name": {
+				Description: "The name of the holiday.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"description": {
+				Description: "A description of the holiday.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"ical_url": {
+				Description:  "A URL to an iCal file containing all occurrences of the holiday.",
+				Type:         schema.TypeString,
+				Optional:     true,
+				RequiredWith: []string{"ical_timezone"},
+				ValidateFunc: validation.IsURLWithHTTPorHTTPS,
+				AtLeastOneOf: []string{"custom_periods", "ical_url"},
+			},
+			"ical_timezone": {
+				Description:  "The timezone to use for events in the iCal file pointed to by ical_url.",
+				Type:         schema.TypeString,
+				Optional:     true,
+				RequiredWith: []string{"ical_url"},
+				ValidateFunc: func(i interface{}, k string) (_ []string, errors []error) {
+					_, err := time.LoadLocation(i.(string))
+					if err != nil {
+						errors = append(errors, fmt.Errorf("expected %q to be a valid IANA Time Zone, got %v: %+v", k, i, err))
+					}
+					return
+				},
+			},
+			"custom_periods": {
+				Description: "A list of custom periods for the holiday.",
+				Type:        schema.TypeList,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: "The name of the custom period.",
+						},
+						"start_time": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.IsRFC3339Time,
+						},
+						"end_time": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.IsRFC3339Time,
+						},
+					},
+				},
+				Optional: true,
+			},
+		},
+	}
+}
+
+func resourceMachineLearningHolidayCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	c := meta.(*client).mlapi
+	holiday, err := makeMLHoliday(d)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	holiday, err = c.NewHoliday(ctx, holiday)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	d.SetId(holiday.ID)
+	return resourceMachineLearningHolidayRead(ctx, d, meta)
+}
+
+func resourceMachineLearningHolidayRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	c := meta.(*client).mlapi
+	holiday, err := c.Holiday(ctx, d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
```

Comment:
There should be a case where the holiday is missing (404). In that case, we remove the resource from state (`d.SetId("")`) and print out a warning

Reply:
Done in a311cf7 (for ML jobs as well).